Performance update

ironage opened the pull request:
This updates the performance reports:
- Adds a couple new tags released recently
- Adds support for comments and empty lines in the `revs_to_benchmark.txt` file
- Adds comparison of the base branch at the point where the current branch diverged

ironage commented:
@finnschiermer I notice a time decrease due to some change between v2.3.1 and v2.3.2 for the benchmark-crud [operations](https://ci.realm.io/job/realm/job/realm-core/job/PR-2514/3/Performance_Report/report.html#read_random_compact_EncryptionOff) 

Do you think it can be explained by this [commit](https://github.com/realm/realm-core/commit/393836fcefcd705f0e197d720eff6ef4744d9c31)? Or is it likely a bug in the performance infrastructure?

emanuelez commented on Jenkinsfile:
```diff
@@ -305,12 +305,12 @@ def buildPerformance() {
           sh 'rm core-benchmarks.zip'
 
-          sh '''
+          sh """
```
Why? There's no string interpolation being done.

danielpovlsen commented on Jenkinsfile:
```diff
@@ -305,12 +305,12 @@ def buildPerformance() {
             ./gen_bench_hist.sh
             ./parse_bench_hist.py --local-html results/ core-benchmarks/
-          '''
+          """
```
If I understand this correctly you are changing the quoting from strong to weak, but there is seemingly nothing to be interpreted inside, so wouldn't it be better to keep it strong? Maybe a n00b question. 😄

danielpovlsen replied:
Sorry, this is Groovy so the `"""` and `'''` are called weak and strong as in shell. The correct term is _interpolated_, but the idea is the same.

emanuelez commented on test/bench/gen_bench_hist.sh:
```diff
@@ -36,19 +40,35 @@ while [ $# -gt 0 ]; do
+  forked=$(git merge-base "${destination}" HEAD)
+  ret=$?
+  if [ $ret -gt 0 ]; then
+      echo "Error: could not find where this branch forked from ${destination} continuing."
```
Is there a dot missing after `${destination}` ?

ironage replied:
Added.

danielpovlsen commented:
> But why is this not targeting master?

Good point. It's a very internal, "non-code" change.

ironage commented:
@emanuelez @danielpovlsen 
I've rebased onto master, good idea.

For some reason I didn't commit the part where Jenkins uses a variable, that's the whole reason I changed quote styles before. But the variable is there now (force pushed because it should have been part of the last commit). I'm pretty sure I need those double quotes in this case, though if there is a better way do tell.

finnschiermer commented:
@ironage: Interesting. The performance increase is consistent and quite large. It's also there in workloads where I wouldn't expect many calls to remap. But my intuition about that could be wrong. If we want to find out, we could profile the benchmarks before and after the change.
